components/admin_console/cluster_settings.jsx

```diff
@@ -211,6 +213,24 @@ export default class ClusterSettings extends AdminSettings {
                     onChange={this.overrideHandleChange}
                     setByEnv={this.isSetByEnv('ClusterSettings.UseExperimentalGossip')}
                 />
+                <BooleanSetting
+                    id='EnableExperimentalGossipEncryption'
+                    label={
+                        <FormattedMessage
+                            id='admin.cluster.EnableExperimentalGossipEncryption'
+                            defaultMessage='Enable Experimental Gossip encryption:'
+                        />
+                    }
+                    helpText={
+                        <FormattedHTMLMessage
+                            id='admin.cluster.EnableExperimentalGossipEncryptionDesc'
+                            defaultMessage='When true, all communication through the gossip protocol will be encrypted.'
+                        />
+                    }
+                    value={this.state.EnableExperimentalGossipEncryption}
+                    onChange={this.overrideHandleChange}
+                    setByEnv={this.isSetByEnv('ClusterSettings.EnableExperimentalGossipEncryption')}
+                />
```

Comment:
Please add Cypress test, specifically for this new setting. Minimum test could be check/uncheck and save the setting, and then see if it's saved into the server by getting the config (see [related test](https://github.com/mattermost/mattermost-webapp/blob/master/e2e/cypress/integration/system_console/ui_and_api/customization_spec.js)).  Let me know if you need help. Thanks.
(In the future, we might going to add test that query directly into the DB and inspect whether the secret key has been generated after enabling the setting.)

Reply:
Oh my, back to JS again then :sweat_smile:

Reply:
Done